### ISSUE TITLE
Enhance front-end build system

### ISSUE THREAD

**Issue by simpsonw** [ualibraries/500#4]
We have the basic skeleton for our front-end build system, now let's flesh it out a bit more.  This task encompasses:

- Adding a watch task for the CSS
- Adding an auto reload plugin (possibly the Browserify livereload plugin)
- Adding a .editorconfig so that syntax highlighting works
- Adding a CSS minifier
- Auditing the packages and removing any unnecessary ones

**Comment by liquid06** [comment]
While installing the theme on @RuiQiDev's computer, we noticed that gulp had to be installed globally in order to run the tasks. In other projects we're able to have a locally-installed gulp and use `npm run gulp` to run it. I'm not sure how to get that set up here, but it seems best to avoid requiring global dependencies. We could investigate that for this ticket if that seems helpful - it doesn't seem urgent but I wanted to add a note about it. :sweat_smile:

**Comment by simpsonw** [comment]
Woops, glad you caught that, @liquid06.  It should be work without a global installation of Gulp via `npm run gulp` now.

**Comment by simpsonw** [comment]
@cg433n helped @liquid06 and I change the build tools over from using Gulp to using Webpack.  For the most part it seems like everything is working great but there's a problem with Browsersync, the auto-reload plugin we're using.  Browsersync works by injecting a `<script>` tag that in turn injects a `<script async>` tag that handles the reloading.  The problem is, for some reason the injection is failing.  I don't know if it has something to do with Omeka or what, but that tag isn't showing up.  I noticed that there's a [Drupal plugin for Browsersync](https://www.drupal.org/project/browsersync) that manually inserts the script tag in the page via a hook.  I'm guessing that whatever limitation that keeps Omeka from working correctly with Browsersync also affects Drupal (perhaps something about the proxy settings and dynamically generated pages?).

In any case, as a proof of concept I added:
```html
<script id="__bs_script__">
  document.write("<script async src='//{{ host }}:{{ port }}/browser-sync/browser-sync-client.js'><\/script>".replace("HOST", location.hostname));
</script>
```
to common/header.php and went to `http://localhost:3000/online-exhibits/live/exhibits/show/reformation/intro` in my browser and it seemed to work, but it's worth discussing if there's some way we can streamline things.

**Comment by simpsonw** [comment]
It turns out that Firefox (which I still have as my default browser) was somehow ignoring the port for Browsersync.  So I would attempt to go to `http://localhost:3000/online-exhibits/live` and Firefox would just send me to `http://localhost/online-exhibits/live`, which wouldn't have the correct JavaScript snippet injected in it.  I set the `open` property to `false` so that Browsersync doesn't automatically open a browser window.  At this point, all of the relevant changes have been merged to master, so I think this task is essentially done.  We should sit down with @RuiQiDev on Monday and work on integrating her work from #5 into the new build system.
